qpmodel/LogicCard.cs

```diff
@@ -157,6 +157,25 @@ public override ulong LogicAggCE(LogicAgg node)
         //
         public override ulong LogicJoinCE(LogicJoin node)
         {
+            ulong getDistinct(Expr key)
+            {
+                if (key is ColExpr col)
+                {
+                    var tr = col.tabRef_;
+
+                    if (tr is FromQueryRef fqr && fqr.MapOutputName(col.colName_) != null)
+                        if (fqr.MapOutputName(col.colName_) is ColExpr ce) tr = ce.tabRef_;
```

Comment:
does this fix change any unittests results? if not, I'd like to add one tests under TestJoin test.